Source/WebCore/platform/audio/gstreamer/AudioSourceProviderGStreamer.cpp

```diff
@@ -90,18 +91,33 @@ AudioSourceProviderGStreamer::AudioSourceProviderGStreamer()
 
 #if ENABLE(MEDIA_STREAM)
 AudioSourceProviderGStreamer::AudioSourceProviderGStreamer(MediaStreamTrackPrivate& source)
-    : m_notifier(MainThreadNotifier<MainThreadNotification>::create())
+    : m_captureSource(source)
+    , m_notifier(MainThreadNotifier<MainThreadNotification>::create())
 {
     initializeDebugCategory();
-    auto pipelineName = makeString("WebAudioProvider_MediaStreamTrack_", source.id());
+    registerWebKitGStreamerElements();
+    const char* pipelineNamePrefix = "";
+#if USE(GSTREAMER_WEBRTC)
+    if (m_captureSource->source().isIncomingAudioSource())
+        pipelineNamePrefix = "incoming-";
+#endif
+    auto pipelineName = makeString(pipelineNamePrefix, "WebAudioProvider_MediaStreamTrack_", source.id());
     m_pipeline = gst_element_factory_make("pipeline", pipelineName.utf8().data());
     GST_DEBUG_OBJECT(m_pipeline.get(), "MediaStream WebAudio provider created");
-    auto src = webkitMediaStreamSrcNew();
-    webkitMediaStreamSrcAddTrack(WEBKIT_MEDIA_STREAM_SRC(src), &source, true);
+
+    m_streamPrivate = MediaStreamPrivate::create(source.logger(), { source });
 
     m_audioSinkBin = gst_parse_bin_from_description("tee name=audioTee", true, nullptr);
 
-    auto* decodebin = makeGStreamerElement("decodebin3", nullptr);
+    auto* decodebin = makeGStreamerElement("uridecodebin3", nullptr);
+
+    g_signal_connect_swapped(decodebin, "source-setup", G_CALLBACK(+[](AudioSourceProviderGStreamer* provider, GstElement* sourceElement) {
+        ASSERT(WEBKIT_IS_MEDIA_STREAM_SRC(sourceElement));
+        if (!WEBKIT_IS_MEDIA_STREAM_SRC(sourceElement))
```

Comment:
Instead of `ASSERT` and then checking and bailing out, you can just `ASSERT_NOT_REACHED` inside the if.